x-pack/plugins/observability_solution/logs_explorer/public/utils/convert_discover_app_state.ts

```diff
@@ -58,18 +75,34 @@ export const getDiscoverAppStateFromContext = (
   filters: cloneDeep(displayOptions.filters),
 });
 
+export const getDiscoverColumnsWithFallbackFieldsFromDisplayOptions = (
+  displayOptions: DisplayOptions
+): DiscoverAppState['columns'] =>
+  displayOptions.grid.columns.flatMap((column) => {
+    if (column.type === 'document-field') {
+      return column.field;
+    } else {
+      return SMART_FALLBACK_FIELDS[column.name];
+    }
+  });
+
 export const getDiscoverColumnsFromDisplayOptions = (
   displayOptions: DisplayOptions
-): DiscoverAppState['columns'] => displayOptions.grid.columns.map(({ field }) => field);
+): DiscoverAppState['columns'] =>
+  displayOptions.grid.columns.flatMap((column) => {
+    return column.type === 'document-field' ? column.field : column.name;
```

Comment:
nit: This logic is repeating several times from what I see: will it make sense to have a function and reuse it?

Reply:
I agree the logic repeats in various places in this PR, but the challenge is with circular dependency. The way these packages and plugins exist, there is no common place where this can be defined and used without having circular dependency problems